turbopack/crates/turbopack-ecmascript/src/analyzer/imports.rs

```diff
@@ -209,8 +212,14 @@ pub(crate) struct ImportMapReference {
 }
 
 impl ImportMap {
-    pub fn is_esm(&self) -> bool {
-        self.has_exports || self.has_imports || self.has_top_level_await
+    pub fn is_esm(&self, specified_type: SpecifiedModuleType) -> bool {
+        match specified_type {
+            SpecifiedModuleType::Automatic => {
+                self.has_exports || self.has_imports || self.has_top_level_await
+            }
+            SpecifiedModuleType::CommonJs => false,
+            SpecifiedModuleType::EcmaScript => true,
+        }
```

Comment:
Moving this logic out of the callsite in `lib.rs` and `references/mod.rs` seems like a very good cleanup 👍, otherwise this API seems like a very subtle footgun (if you don't check `SpecifiedModuleType`, it will work 90% of the time until it doesn't).